src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -343,6 +343,8 @@ private function createTypeAlternatives(TypedReference $reference)
             $message = sprintf('the existing "%s" service', $this->types[$type]);
         } elseif ($reference->getRequiringClass() && !$reference->canBeAutoregistered()) {
             return ' It cannot be auto-registered because it is from a different root namespace.';
+        } elseif (is_array($this->types) && (false !== $key = array_search(strtolower($type), array_map('strtolower', $this->types)))) {
+            return sprintf(' Do you mean "%s" ?', $this->types[$key], $type);
```

Comment:
What about: `Maybe the typehint has a typo and should refer to "%s" instead?` ?

Reply:
Based on @nicolas-grekas' comment above... we can't change this to mention the typehint as it would also work for named services actually.